ts/api/api.ts

```diff
@@ -185,109 +205,136 @@ class OpenYoloApiImpl implements OpenYoloApi {
         providerUrlBase,
         preloadRequest);
 
-    let channel: SecureChannel|null = null;
-    if (areTimeoutsDisabled) {
-      channel = await SecureChannel.clientConnectNoTimeout(
-          window, frameManager.getContentWindow(), instanceId, instanceIdHash);
-    } else {
-      channel = await SecureChannel.clientConnect(
-          window, frameManager.getContentWindow(), instanceId, instanceIdHash);
-    }
+    const channel =
+        await timeoutRacer.race(SecureChannel.clientConnectNoTimeout(
+            window,
+            frameManager.getContentWindow(),
+            instanceId,
+            instanceIdHash));
 
-    // Check whether the client should wrap the browser's navigator.credentials.
+    // Check whether the client should wrap the browser's
+    // navigator.credentials.
     const request = new WrapBrowserRequest(frameManager, channel);
-    const timeoutMs =
-        areTimeoutsDisabled ? undefined : DEFAULT_TIMEOUTS.wrapBrowserRequest;
-    const wrapBrowser =
-        await request.dispatch(undefined, timeoutMs).catch((error) => {
-          // Ignore errors.
-          return false;
-        });
+    let wrapBrowser = false;
+    try {
+      wrapBrowser = await request.dispatch(undefined, timeoutRacer);
+    } catch (e) {
+      // Only rethrow a timeout error.
+      timeoutRacer.rethrowTimeoutError(e);
```

Comment:
Right, I should just ignore here. I added a logic to reset the instance though if the init fails, so that one timeout will not prevent subsequent calls to the API.